src/TypeParser/PhpTypeParser.php

```diff
@@ -20,12 +20,18 @@ final class PhpTypeParser
     private const TYPE_MIXED = 'mixed';
     private const TYPE_RESOURCE = 'resource';
     private const TYPE_ARRAY = 'array';
+    private const TYPE_ARRAY_COLLECTION = 'ArrayCollection';
+    private const TYPE_COLLECTION = 'Collection';
```

Comment:
why do you support collection here but not in the jms parser?

should we do Collection::class everywhere and use is_subclass_of to handle inheritance correctly? (also this could clash with some other class called Collection if we don't use the namespace)

as all collections have ArrayAccess, i think it is correct to support generic Collection. afaik doctrine itself works like this, where you put an ArrayCollection on an entity field, and when you load from the database you get some other type of collection that dynamically loads the referenced entities on demand.

Reply:
I know check for the actual resolved class via d4801f965e1aa72c2b70d4d8c9b943acb3fb1e49 - This makes sure that only collection classes from doctrine are treated as such.